main/src/ca/uwaterloo/flix/api/UnitClass.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 Ramin Zarifi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.api;
+
+/**
+ * UnitClass which is used to represent `Unit`
+ */
+public class UnitClass {
```

Comment:
I would prefer if we just use the names:

- `UnitClass -> Unit`
- `TagInterface -> Enum`
- `TupleInterface -> Tuple`.

The interface/class thing is not really idiomatic Java, Scala, or Flix.